doc/service-tests.md

```diff
@@ -7,255 +7,249 @@ automated tests should be included. They serve three purposes:
 refer to their [API reference][Joi API].
-
-Since we don't know whether rust will be passing or not at the time the test
-runs, we use `Joi.equal()`, which accepts any of the values passed in.
-
-Notice we don't have to specify `/travis` again, or even `localhost`. The test
-runner handles that for us.
+4. `Joi.object().keys()` defines a Joi object schema containing some defined keys
+5. We expect `name` to be a string literal `"build"`
+6. Because this test depends on a live service, we don't want our test to depend on our API call returning a particular build status. Instead we should perform a "picture check" to assert that the badge data conforms to an expected pattern. Our test should not depend on the status of the example project's build, but should fail if trying to generate the badge throws an error, or if there is a breaking change to the upstream API. In this case we will use a pre-defined regular expression to check that the badge value looks like a build status. [services/test-validators.js](https://github.com/badges/shields/blob/master/services/test-validators.js) defines a number of useful validators we can use. Many of the common badge types (version, downloads, rank, etc) already have validators defined here.
 
 When defining an IcedFrisby test, typically you would invoke the `toss()`
 method, to register the test. This is not necessary, because the Shields test
 harness will call it for you.
 
-[Rust]: https://www.rust-lang.org/en-US/
 [IcedFrisby]: https://github.com/MarkHerhold/IcedFrisby
 [longer example]: https://github.com/MarkHerhold/IcedFrisby/#show-me-some-code
 [IcedFrisby API]: https://github.com/MarkHerhold/IcedFrisby/blob/master/API.md
 [Joi]: https://github.com/hapijs/joi
 [Joi API]: https://github.com/hapijs/joi/blob/master/API.md
 
-Run the test:
+### (3) Running the Tests
+
+Lets run the test we have written:
 
 ```
-npm run test:services -- --only=travis
+npm run test:services -- --only=wercker
 ```
 
 The `--only=` option indicates which service or services you want to test. You
-can provide a comma-separated list of ids.
-These ids are those passed to the ServiceTester:
-
-```
-... ServiceTester({ id: 'travis', ... })
-```
+can provide a comma-separated list here.
 
 The `--` tells the NPM CLI to pass the remaining arguments through to the test
 runner.
 
 Here's the output:
 
 ```
-http://localhost:1111/
-  Travis CI
-    build status on default branch
+Server is starting up: http://lib/service-test-runner/cli.js:80/
+  Wercker
+    Build status
       ✓
-  [ GET http://localhost:1111/travis/rust-lang/rust.json ] (265ms)
-
+        [ GET /build/wercker/go-wercker-api.json ] (572ms)
 
   1 passing (1s)
 ```
 
 That's looking good!
 
-Next we'll add a second test for a branch build.
+Sometimes if we have a failing test, it is useful to be able to see some logging output to help work out why the test is failing. We can do that by calling `npm run test:services:trace`. Try running
```

Comment:
What do you think about adding a info how to run one test? I use this command: `npm run test:services -- --only=wercker --fgrep="Build passed (mocked)"`

Reply:
I never knew this existed until today. Every day's a school day :school: